### PR TITLE
feat: XRPL AI

### PR DESCRIPTION
**Pull request by justinr1234**
- Adds a chatbot that can answer questions about the xrpl.org docs set
- Wiki with backgroud information about the project: https://github.com/XRPLF/xrpl-dev-portal/wiki/XRPL-AI
- Working preview link: https://xpring-eng.github.io/xrplai-js/

### CODE REVIEW

File: template/base.html.jinja [hunk 1]
```diff
@@ -147,6 +148,94 @@ $(document).ready(() => {
...
+          name: 'algolia',
+          apiKey: '3431349deec23b0bc3dcd3424beb9a6e',
+          appId: 'R39QY3MZC7',
+          indexName: 'xrpl',
+        },
+      }
+    },
+  };
+</script>
+<script id="initjs" type="module" src="https://www.unpkg.com/@xrpl/ai-web/dist/init.js"></script>
```
Comment (jonathanlei) [inline template/base.html.jinja:239]: are you using the init script locally or from www.unpkg.com? 
if you are using the hosted version then no need to add `assets/js/xrplai.js`?

Comment (justinr1234) [reply template/base.html.jinja:239]: Those snuck in there, I've removed them. We'll be using the hosted version from npm

**Comment by WietseWind** [comment]
@justinr1234 Initial review: looks good, but the action button is hidden behind the "scroll up" icon & cookie icon:

image

**Comment by jonathanlei** [comment]
Screenshot 2023-10-10 at 1 37 36 PM

Also the link to privacy policy in light mode is the same color as the background

**Comment by justinr1234** [comment]
> @justinr1234 Initial review: looks good, but the action button is hidden behind the "scroll up" icon & cookie icon:
> 
> image

I made the fix, it will show up on the preview after we merge this code to main

**Comment by justinr1234** [comment]
> Screenshot 2023-10-10 at 1 37 36 PM
> Also the link to privacy policy in light mode is the same color as the background

Fixed, but it won't show up on the preview link

**Review by amarantha-k — APPROVED** [review]
+1 to publish this as a Beta/preview and gather user feedback. 
The Docs team has submitted feedback on the (content) responses and AI behavior separately and we understand that the feedback will be addressed in the next phase.

**Comment by jonathanlei** [comment]
Screenshot 2023-10-10 at 3 01 23 PM
please ignore this if you fixed this already, the feedback module/texts are off on light mode

**Comment by justinr1234** [comment]
> Screenshot 2023-10-10 at 3 01 23 PM please ignore this if you fixed this already, the feedback module/texts are off on light mode

Fixed

**Comment by ckniffen** [comment]
I was unable to see the prompt feedback text when in dark mode. 

Monosnap Home - XRPL org 2023-10-10 20-51-06

**Comment by justinr1234** [comment]
> I was unable to see the prompt feedback text when in dark mode.
> 
> Monosnap Home - XRPL org 2023-10-10 20-51-06

Weird the background of the textbox is actually a different color than on your screenshot. I just changed it to black text on a white background, should make it so everyone gets the same thing.

**Comment by tequdev** [comment]
The back of the modal is darkened, but the Card and header areas are unchanged and feel strange.
スクリーンショット 2023-10-11 14 05 46

**Comment by Silkjaer** [comment]
Changing to dark mode, the modal remains white, however some of the text in the modal is white as well.

image

**Comment by justinr1234** [comment]
> * https://xpring-eng.github.io/xrplai-js/

I'm actually not sure how you have a white modal. The modal is supposed to always have a black background. Can you provide more information from the browser dev tools?

**Comment by justinr1234** [comment]
> Changing to dark mode, the modal remains white, however some of the text in the modal is white as well.
> 
> image

The modal should never be white to begin with, so that's super weird to me. Do you have any further information from the dev cosnole?

**Comment by justinr1234** [comment]
Nevermind I see, if you change your system settings to light mode it makes the modal white

**Comment by justinr1234** [comment]
> The back of the modal is darkened, but the Card and header areas are unchanged and feel strange. スクリーンショット 2023-10-11 14 05 46

Fixed

**Comment by justinr1234** [comment]
> Changing to dark mode, the modal remains white, however some of the text in the modal is white as well.
> 
> image

Fixed

**Comment by Silkjaer** [comment]
On mobile (iOS 17) the modal is a bit too tall. Other than that I’m good to approve. 
IMG_3729

**Comment by justinr1234** [comment]
> Changing to dark mode, the modal remains white, however some of the text in the modal is white as well.
> 
> image

Fixed

**Comment by Silkjaer** [comment]
> On mobile (iOS 17) the modal is a bit too tall. Other than that I’m good to approve.

turns out this is a site-wide issue unrelated to this PR. I’m good to approve this as is

**Review by Silkjaer — APPROVED** [review]

**Comment by amarantha-k** [comment]
Not sure if this was fixed, but in dark mode, still seeing text (links) in white.  
Screenshot 2023-10-11 at 9 07 29 AM

**Comment by jscottbranson** [comment]
Overall, I think this looks good! The only feedback I have at this point is:

1. When I respond to "Was this response helpful?", the page does not scroll down, so unless there is already empty space after the "Yes"/"No" buttons, it's easy to miss the additional feedback survey question. I would suggest having the page scroll down a little bit automatically after "Yes" or "No" is selected.

1. If a user selects "No" under "Was this response helpful?" the follow up question should not be the same as when "Yes" is selected.

AI screenshot

**Pull request merged**